theano/tensor/nnet/tests/test_nnet.py

```diff
@@ -1437,7 +1454,7 @@ def test_basic(self):
         # test that function contains softmax and no div.
         f = theano.function([c], p_y, mode=self.mode)
 
-        assert hasattr(f.maker.fgraph.outputs[0].tag, 'trace')
+        assert check_stack_trace(f, ops_to_check=softmax_op)
```

Comment:
In that case, the softmax does not have a stack trace. This is probably something to fix in the optimization, or maybe in another optimization it depends on.